packages/widget/src/widget.ts

```diff
@@ -1,14 +1,37 @@
 import type { HTMLTemplateResult } from 'lit';
 import { LitElement, html } from 'lit';
-import { customElement } from 'lit/decorators.js';
+import { customElement, property } from 'lit/decorators.js';
 import './components/widget-app';
+import type { ethers } from 'ethers';
+import type { ApiPromise } from '@polkadot/api';
 import { styles } from './styles';
 import { Network } from './controllers';
+import type { ISygmaProtocolWidget } from './interfaces';
 
 @customElement('sygmaprotocol-widget')
 class SygmaProtocolWidget extends LitElement {
   static styles = styles;
 
+  @property({
+    type: Object
+  })
+  network?: Network;
+
+  @property({
+    type: Object
+  })
+  web3Provider?: ethers.providers.Web3Provider;
+
+  @property({
+    type: Object
+  })
+  apiPromise?: ApiPromise;
```

Comment:
maybe rename to "substrateProvider"? I would also merge ApiPromise and wssConnectionUrl into single property

Reply:
If I understand you correctly, susbtrateProvider could be either ApiPromise or string?

Reply:
the goal is to enable dapp to provider substrate related stuff as object in single property. So dapp should be able to, optionally, provide signer and provider.
So the type could be:
```typescript
interface {
  provider?: ApiPromise|string,
  signer: NotSure //it needs to support external signer like extension
}
˙``

Reply:
thibaut says:
`˙`
api: ApiPromise | string
signer: Signer // that's actually a type from '@polkadot/api/types'
senderAddress: string // the signer doesn't have the address easy to get afaics
```

Reply:
how the user flow is going to work passing signer or sender address -which you get for instance interacting with the dot extension- considering that we select network and then connect wallet? If some of those entities that reflect actual connection or capacity to sign transactions are provided, is the `from network` selector going to be prefilled?

Reply:
I think it makes sense to prefill from network if website provided api.
In theory, maybe it makes sense to only allow providing signer and we handle creating provider, at least for substrate, wdyt?

Reply:
for substrate makes a lot of sense to just provide signer, since signer and provider are somewhat decouple in a way. What about ethereum? Here we give the option to pass web3Provider just because is being used on the SDK. I guess for MVP purposes makes sense to go along with that definition